### DISCUSSION TITLE
Marquis 2019 SPMe

### DISCUSSION THREAD

**Discussion by gishan12** [pybamm-team/PyBaMM#2076] [Q&A] [answered]
Hi, I was going through the Marquis 2019 paper and trying to replicate its SPMe. However, I realised that the full diffusion model in Pybamm uses equations 3a - 3i [which is the SPMe(S) model]. But, I want to use the canonical version of this, or plain SPMe which is give by equations 40a to 40p and I am not sure how to use this version of SPMe on Pybamm. I am guessing I need to specify the type of diffusion in the options as maybe 'first order' or 'leading order' but I am not sure. Any insight ifs much appreciated.

**Comment by valentinsulzer** [comment, marked as answer]
The SPMe in pybamm is the canonical SPMe (40a-p), except using non-linear diffusion instead of linear diffusion for (40c) (see "Timms, R., et al. "Corrigendum to “Generalised single particle models for high-rate operation of graded lithium-ion electrodes: Systematic derivation and validation”[Electrochimica Acta 339 (2020) 135862]." Electrochimica Acta 351 (2020).")

It does use the "full diffusion" submodel, but `i_e` is specified as the piecewise linear function by the composite electrolyte conductivity model (instead of the full conductivity model), and `j` is piecewise constant, so the resulting diffusion equation is different from DFN.

**Reply by Saketh1729** [reply]
Hi, I am trying to reconstruct the SPMe model using PyBAMM's base model to learn PyBAMM in further detail, can you please point me to where I could find the code modelling the equations governing SPMe. I have gone through full_diffusion submodel and was wondering if there was a more direct approach to modelling the electrolyte dynamics.
